src/MainWindow.vala

```diff
@@ -25,42 +25,53 @@ public class ShortcutOverlay.MainWindow : Gtk.Window {
     }
 
     construct {
-        var settings_button = new Gtk.Button () {
-            icon_name = "preferences-system-symbolic",
-            tooltip_text = _("Keyboard Settings…")
-        };
-
-        var shortcuts_view = new ShortcutsView () {
+        var box = new Gtk.Box (Gtk.Orientation.VERTICAL, 24) {
             margin_start = 36,
             margin_end = 36,
             margin_top = 12,
-            margin_bottom = 36
+            margin_bottom = 24
         };
 
-        child = shortcuts_view;
+        var shortcuts_view = new ShortcutsView ();
 
-        var start_controls = new Gtk.WindowControls (Gtk.PackType.START);
+        var settings_button = new Gtk.Button.with_label (_("Keyboard Settings…")) {
+            halign = Gtk.Align.END
+        };
+
+        box.append (shortcuts_view);
+        box.append (settings_button);
+        child = box;
+
+        var start_controls = new Gtk.WindowControls (Gtk.PackType.START) {
+            hexpand = true,
+            halign = Gtk.Align.START
+        };
+        var end_controls = new Gtk.WindowControls (Gtk.PackType.END) {
+            hexpand = true,
+            halign = Gtk.Align.END
+        };
+
+        var controls_box = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
+        controls_box.append (start_controls);
+        controls_box.append (end_controls);
```

Comment:
Since we've got just a standard header layout now, can we use a regular Gtk.HeaderBar with `Granite.STYLE_CLASS_FLAT` and `Granite.STYLE_CLASS_DEFAULT_DECORATION` instead of creating a custom titlebar?